src/Shared/Traits.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//-----------------------------------------------------------------------
+// </copyright>
+
+using System;
+
+namespace Microsoft.Build.Utilities
+{
+    /// <summary>
+    ///     Represents toggleable features of the MSBuild engine
+    /// </summary>
+    internal class Traits
+    {
+        public static Traits Instance = new Traits();
+
+        public EscapeHatches EscapeHatches { get; }
+
+        public Traits()
+        {
+            EscapeHatches = new EscapeHatches();
+        }
+    }
+
+    internal class EscapeHatches
+    {
+        public bool SerializeEntireProjectInstance => string.IsNullOrEmpty(Environment.GetEnvironmentVariable("MSBUILDDONOTSERIALIZEENTIREPROJECTINSTANCE"));
```

Comment:
Never mind, read that wrong :)

---
In reply to: [113763543](https://github.com/Microsoft/msbuild/pull/2018#discussion_r113763543) [](ancestors = 113763543)